app/shop.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+from app.customer import Customer
+
+
+class Shop:
+    def __init__(self, name: str, location: list, products: dict) -> None:
+        self._name = name
+        self._location = location
+        self._products = products
+
+    @classmethod
+    def create_shop(cls, shop_info: dict) -> Shop:
+        return cls(
+            shop_info["name"],
+            shop_info["location"],
+            shop_info["products"]
+        )

```

Comment:
The `create_shop` method assumes that the dictionary `shop_info` will always contain the keys 'name', 'location', and 'products'. It would be safer to add error handling to manage cases where these keys might be missing, such as using `shop_info.get('name')` with a default value or raising an informative exception.